pkg/helpers/helpers.go

```diff
@@ -887,11 +916,126 @@ func FeatureGatesArgs(featureGates []operatorapiv1.FeatureGate, component string
+// to the status of the cluster manager or klusterlet
+func CheckRegistrationFeatureGates(operatorType string,
+	configuration *operatorapiv1.RegistrationConfiguration) ([]string, metav1.Condition, error) {
+	if operatorType != OperatorTypeClusterManager && operatorType != OperatorTypeKlusterlet {
+		return nil, metav1.Condition{}, fmt.Errorf("not supported operatorType: %s", operatorType)
+	}
+
+	if configuration == nil {
+		return checkNilFeatureGates(operatorType, componentNameRegistration)
+	}
+	return checkFeatureGates(operatorType, componentNameRegistration, configuration.FeatureGates)
+}
+
+// CheckWorkFeatureGates check if feature gates are valid for WorkConfiguration, return
+// the valid feature gates, the condition and if there is error, this func will not
+// update the condition to the status of the cluster manager or klusterlet
+func CheckWorkFeatureGates(operatorType string,
+	configuration *operatorapiv1.WorkConfiguration) ([]string, metav1.Condition, error) {
+	if operatorType != OperatorTypeClusterManager && operatorType != OperatorTypeKlusterlet {
+		return nil, metav1.Condition{}, fmt.Errorf("not supported operatorType: %s", operatorType)
+	}
+
+	if configuration == nil {
+		return checkNilFeatureGates(operatorType, componentNameWork)
+	}
+	return checkFeatureGates(operatorType, componentNameWork, configuration.FeatureGates)
+}
+
+func checkNilFeatureGates(operatorType string, componentName string) ([]string, metav1.Condition, error) {
+	conditionType := getConditionType(componentName)
+	componentKey := getComponentKey(operatorType, componentName)
+
+	return defaultNilFeatureGates(componentKey), metav1.Condition{
+		Type:    conditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  FeatureGatesReasonNil,
+		Message: fmt.Sprintf("%s feature gates configuration is nil ", componentName),
+	}, nil
+}
+
+func checkFeatureGates(operatorType string, componentName string,
+	featureGates []operatorapiv1.FeatureGate) ([]string, metav1.Condition, error) {
+
+	conditionType := getConditionType(componentName)
+	componentKey := getComponentKey(operatorType, componentName)
+
+	if len(featureGates) > 0 {
+		featureGateArgs, invalidFeatureGates := FeatureGatesArgs(featureGates, componentKey)
+		if len(invalidFeatureGates) == 0 {
+			return featureGateArgs, metav1.Condition{
+				Type:    conditionType,
+				Status:  metav1.ConditionTrue,
+				Reason:  FeatureGatesReasonAllValid,
+				Message: fmt.Sprintf("%s feature gates are all valid", componentName),
+			}, nil
+		}
+
+		// for the invalid feature gates, we show them in the condition, here only return the valid
+		// feature gates, so the caller can use the valid ones and degrade the invalid ones to the
+		// default values
+		return featureGateArgs, metav1.Condition{
+			Type: conditionType, Status: metav1.ConditionFalse, Reason: "InvalidFeatureGatesExisting",
+			Message: fmt.Sprintf("there are some invalid feature gates of %s: %v, will process them with default values",
```

Comment:
if the feature gate is invalid, will use the default value to deploy the registration/work-agent, add an invalid condition.
@qiujian16 Do you think this is reasonable?

Reply:
the condition will be invalid, and print the invalid feature gate. And will use the user-provided value for the correct one and the default value for the invalid one to deploy registration/work agent.